### ISSUE TITLE
Lastest human gtf file breaks RNA-SeQC run

### ISSUE THREAD

**Issue by serine** [broadinstitute/RNA-SeQC#1]
Hi there,

This is the very first time I am using `RNA-SeQC` tool. My colleague did a test run and it worked well for him. I, on the other hand came across a problem.

I’ve raised an issue on the BioStars, here is the link https://www.biostars.org/p/103035/#138553
to see if anyone else had the same issue. Note that this isn’t my question, rather I added a new comment no Friday.

They issue: `RNA-SeQC` complains about no `transcript_id` found in the first line

The line that describes gene feature missing `transcript_id` in the latest .gtf file - GRCh38 ftp://ftp.ensembl.org/pub/release-79/gtf/homo_sapiens/
and gencode.v22
ftp://ftp.sanger.ac.uk/pub/gencode/Gencode_human/release_22/gencode.v22.annotation.gtf.gz

And this is what causing an error in `RNA-SeQC` (at least that is what i nailed it down to). It appears to me that `transcript_id` field was inserted into the line which describes the gene. However both `gene_id` and `transcript_id` are identical in the gene lines. I’ve checked several gtf files:
- `gencode.v19.annotation.gtf`
- `gencode.v20.annotation.gtf`
- `gencode.v21.annotation.gtf`
- `gencode.v22.annotation.gtf`
- `gencode.v7.annotation.gtf`
- `Homo_sapiens.GRCh37.62.gtf`
- `Homo_sapiens.GRCh38.79.gtf`

And these two, `gencode.v22.annotation.gtf` and `Homo_sapiens.GRCh38.79.gtf`, don’t have that `transcript_id` field in the gene line.

When my colleague run RNA-SeQC he used gencode.v7.annotation.gtf ftp://ftp.sanger.ac.uk/pub/gencode/Gencode_human/release_7/gencode.v7.annotation.gtf.gz

I will revert back to the latest stable release i.e gencode.v20.annotation.gtf. However this is rather bizarre in my view that that `transcript_id` field in the gene line is identical to `gene_id` on the same line. So it seems that the latest gtf file has improved their format, but it looks like this breaks `RNA-SeQC`

Regards, 

Kirill

**Comment by karl616** [comment]
Not sure if this repository is alive, but it was one of the first hits I found trying to solve the problem described above.

Duplicating the gene_id tag can be done by:

sed -e '/\tgene\t/ s/gene_id \(.ENSG[0-9]*.[0-9]*..\)/\0 transcript_id \1/' gencode.v22.annotation.gtf > gencode.v22.annotation.RNA-SeQC.gtf

**Comment by plijnzaad** [comment]
@karl616: I think this repo and tool are pretty much dead. I found that the combination of gencode-v22 and your `sed` trick at least didn't crash, but then I ran in to Java problems (see issue #2 )

**Comment by serine** [comment]
for sure, been dead for a while not second life for this tool..

I've been running bunch of different tools. [qualiMap](http://qualimap.bioinfo.cipf.es/) to get intragenic/interegenic rates, bunch of picard subtools, but we summarising everything with [MultiQC](http://multiqc.info/) into single html report.

**Comment by plijnzaad** [comment]
OK, sad, but good to know. Might be useful to put up a notice in the
README, right? Incidentally, the only thing I need RNA-SeQC for currently
is coverage plot relative to the 3'-end of the transcripts.  All other
tools stretch the full transcript to 100%, which is less useful in my
opinion. I'd be grateful for any suggestions on existing (and supported:-)
tools that can do this out of the box, though.

Regards,

Philip

2018-04-10 1:29 GMT+02:00 Kirill Tsyganov <notifications@github.com>:

**Comment by dongjoban17** [comment]
@plijnzaad @serine I know it may be too late to reply to this issue, but I found a post on biostars discussing a similar issue with RNA-SeQC. It doesn't appear to be the same Java problem, but someone was able to resolve the issues by using Java 1.7 instead of 1.8.

I'm currently doing a test run with gencode.v25 and will update if it works.

**Comment by plijnzaad** [comment]
> someone was able to resolve the issues by using Java 1.7 instead of 1.8.

you're right, this is what I do too (I prolly saw the same biostars post). Cheers,

Philip
